eslint: Add accessibility rules

regexowl opened the pull request:
This adds eslint-plugin-jsx-a11y to the eslint configuration, setting up recommended set of rules for accessibility testing.

The error output by `jsx-a11y/no-autofocus` on encountering `autoFocus` attribute was switched to warning for now as whether to auto focus on an element or not can be highly contextual.

Fixes #948

regexowl commented:
The topic of accessibility is very broad so there's a lot to discuss. This PR is supposed to serve as the bare minimum and also a reminder that we might want to look into it a bit deeper.

jkozol reviewed (dismissed):
I think it's great to start getting warnings for missing a11y support!

regexowl commented:
Resolved conflict, rebased and also added a11y plugin to deps as it might have been included in `eslint-config-react-app`. But that was removed in previously merged PR so it didn't work anymore? Not sure what happened there, all should be well now :no_mouth:

regexowl commented:
/retest

regexowl commented:
/retest
